OpenRA.Mods.Common/Traits/Attack/AttackTurreted.cs

```diff
@@ -20,12 +20,14 @@ public class AttackTurretedInfo : AttackFollowInfo, Requires<TurretedInfo>
 		public override object Create(ActorInitializer init) { return new AttackTurreted(init.Self, this); }
 	}
 
-	public class AttackTurreted : AttackFollow
+	public class AttackTurreted : AttackFollow, INotifyCreated
 	{
 		protected Turreted[] turrets;
 
 		public AttackTurreted(Actor self, AttackTurretedInfo info)
-			: base(self, info)
+			: base(self, info) { }
+
+		public void Created(Actor self)
```

Comment:
The `Requires<TurretedInfo>` ensures that the `Turreted` traits have already been initialized before this one, so there should be no problem with keeping this query in the constructor.